egs/voxceleb/v2/prepare_meld.sh

```diff
@@ -0,0 +1,155 @@
+		$DATA_INPUT_DIR 
+		$DATA_OUTPUT_TRAIN_DIR 
+		$DATA_OUTPUT_DEV_DIR
+		$DATA_OUTPUT_TEST_DIR
+	)
+	for dir in "${dirs[@]}"
+	do
+		if [ ! -d $dir ]; then
+			mkdir -p $dir
+			sudo chmod -R 775 $dir
+		fi
+	done
+fi
+
+# prepare reference model (safe to rerun; it will
+# just over-write any modified reference model)
+if [ $stage -eq 1 ]; then
+	# first, we delete everything that's already
+	# in MODEL_OUTPUT_DIR (so we have a fresh start)
+	rm -rf "$MODEL_OUTPUT_DIR/*"
+
+	# first, we generate a new config for our modified
+	# model which modifies the output dimensionality of
+	# the final layer (from # of speakers in vox2 to # of emotions)
+	
+	feat_dim=$NUM_INPUT_DIMENSIONS
+	num_targets=$NUM_TARGET_DIMENSIONS
+
+	# 
+	# START: copied with minimal modification from run_xvector.sh
+	#
+	max_chunk_size=10000
+	min_chunk_size=25
+	mkdir -p $MODEL_OUTPUT_DIR/configs
+	cat <<-EOF > $MODEL_OUTPUT_DIR/configs/modified_nnet.xconfig
+		# please note that it is important to have input layer with the name=input
+
+		# The frame-level layers
+		input dim=${feat_dim} name=input
+		relu-batchnorm-layer name=tdnn1 input=Append(-2,-1,0,1,2) dim=512
+		relu-batchnorm-layer name=tdnn2 input=Append(-2,0,2) dim=512
+		relu-batchnorm-layer name=tdnn3 input=Append(-3,0,3) dim=512
+		relu-batchnorm-layer name=tdnn4 dim=512
+		relu-batchnorm-layer name=tdnn5 dim=1500
+
+		# The stats pooling layer. Layers after this are segment-level.
+		# In the config below, the first and last argument (0, and ${max_chunk_size})
+		# means that we pool over an input segment starting at frame 0
+		# and ending at frame ${max_chunk_size} or earlier.  The other arguments (1:1)
+		# mean that no subsampling is performed.
+		stats-layer name=stats config=mean+stddev(0:1:1:${max_chunk_size})
+
+		# This is where we usually extract the embedding (aka xvector) from.
+		relu-batchnorm-layer name=tdnn6 dim=512 input=stats
+
+		# This is where another layer the embedding could be extracted
+		# from, but usually the previous one works better.
+		relu-batchnorm-layer name=tdnn7 dim=512
+		output-layer name=output include-log-softmax=true dim=${num_targets}
+	EOF
+
+	steps/nnet3/xconfig_to_configs.py \
+		--xconfig-file $MODEL_OUTPUT_DIR/configs/modified_nnet.xconfig \
+		--config-dir $MODEL_OUTPUT_DIR/configs/
+	cp $MODEL_OUTPUT_DIR/configs/final.config $MODEL_OUTPUT_DIR/modified_nnet.config
+
+	# this is for x-vector extraction (in case we need it)
+	echo "output-node name=output input=tdnn6.affine" > $MODEL_OUTPUT_DIR/extract.config
+	echo "$max_chunk_size" > $MODEL_OUTPUT_DIR/max_chunk_size
+	echo "$min_chunk_size" > $MODEL_OUTPUT_DIR/min_chunk_size
+
+	#
+	# END: copied with minimal modification from run_xvector.sh
+	#
+
+	# now, with our updated target config in hand, we copy our
+	# reference model, modifying its final output layer and setting
+	# the learning rates for the first 6 layers to 0
+	nnet3-copy \
+		--nnet-config="$MODEL_OUTPUT_DIR/modified_nnet.config" \
```

Comment:
going through the Kaldi code, it seems like specifying a -nnet-config when copying a nnet will result in kaldi throwing away any layers that are specified in both the base model and the config (preferring whatever is specified in the config) but this doesn't seem to be happening in practice.  when i get around to retraining the modified reference model on emotion, it fails with:
```
ERROR (nnet3-train[5.5.528~1-880df]:ComputeObjectiveFunction():nnet-training.cc:349) Nnet versus example output dimension (num-classes) mismatch for 'output': 3 (nnet) vs. 6 (egs)
```
(where 3 is the number of Vox speakers that I trained the reference model on).  Should I be doing this some other way?
```